packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -84,76 +99,86 @@ function ListViewBlockSelectButton(
 	}
 
 	return (
-		<a
-			className={ clsx(
-				'block-editor-list-view-block-select-button',
-				className
-			) }
-			onClick={ onClick }
-			onContextMenu={ onContextMenu }
-			onKeyDown={ onKeyDown }
-			onMouseDown={ onMouseDown }
-			ref={ ref }
-			tabIndex={ tabIndex }
-			onFocus={ onFocus }
-			onDragStart={ onDragStartHandler }
-			onDragEnd={ onDragEnd }
-			draggable={ draggable }
-			href={ `#block-${ clientId }` }
-			aria-describedby={ ariaDescribedBy }
-			aria-expanded={ isExpanded }
-		>
-			<ListViewExpander onClick={ onToggleExpanded } />
-			<BlockIcon
-				icon={ blockInformation?.icon }
-				showColors
-				context="list-view"
-			/>
-			<HStack
-				alignment="center"
-				className="block-editor-list-view-block-select-button__label-wrapper"
-				justify="flex-start"
-				spacing={ 1 }
-			>
-				<span className="block-editor-list-view-block-select-button__title">
-					<Truncate ellipsizeMode="auto">{ blockTitle }</Truncate>
-				</span>
-				{ blockInformation?.anchor && (
-					<span className="block-editor-list-view-block-select-button__anchor-wrapper">
-						<Badge className="block-editor-list-view-block-select-button__anchor">
-							{ blockInformation.anchor }
-						</Badge>
-					</span>
+		<>
+			<a
+				className={ clsx(
+					'block-editor-list-view-block-select-button',
+					className
 				) }
-				{ isSticky && (
-					<span className="block-editor-list-view-block-select-button__sticky">
-						<Icon icon={ pinSmall } />
-					</span>
-				) }
-				{ images.length ? (
-					<span
-						className="block-editor-list-view-block-select-button__images"
-						aria-hidden
-					>
-						{ images.map( ( image, index ) => (
-							<span
-								className="block-editor-list-view-block-select-button__image"
-								key={ image.clientId }
-								style={ {
-									backgroundImage: `url(${ image.url })`,
-									zIndex: images.length - index, // Ensure the first image is on top, and subsequent images are behind.
-								} }
-							/>
-						) ) }
-					</span>
-				) : null }
-				{ shouldShowLockIcon && (
-					<span className="block-editor-list-view-block-select-button__lock">
-						<Icon icon={ lock } />
+				onClick={ onClick }
+				onDoubleClick={ handleRename }
```

Comment:
```suggestion
				onDoubleClick={ canRemove ? () =>  () => setIsRenameModalOpen( true ) : undefined }
```

We could just inline callback here. The code suggestion probably needs lint fixes.